stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/factory/MediaAttachmentFactory.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.ui.attachments.factory
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.compose.ui.attachments.AttachmentFactory
+import io.getstream.chat.android.compose.ui.attachments.content.MediaAttachmentContent
+import io.getstream.chat.android.compose.ui.attachments.content.MediaAttachmentPreviewContent
+import io.getstream.chat.android.compose.ui.attachments.content.PlayButton
+import io.getstream.chat.android.uiutils.constant.AttachmentType
+
+/**
+ * An [AttachmentFactory] that is able to handle Image and Video attachments.
+ *
+ * @param maximumNumberOfPreviewedItems The maximum number of thumbnails that can be displayed
+ * in a group when previewing Media attachments in the message list.
+ * @param contentPlayButton Displays a play button above video attachments
+ * in the messages list.
+ * @param previewContentPlayButton Displays a play button above video attachments
+ * in the message input.
+ */
+@Suppress("FunctionName")
+public fun MediaAttachmentFactory(
+    maximumNumberOfPreviewedItems: Int = 4,
+    contentPlayButton: @Composable () -> Unit = {
+        DefaultContentPlayButton()
```

Comment:
This approach of passing in content works very well for our lists and so on, but I'm not sure how to apply this to `MediaGalleryPreviewActivity` since it's a separate "island" so to speak.

The only idea I have is including content inside `ChatTheme`, but that feels clunky.